addons/website_sale/controllers/main.py

```diff
@@ -582,6 +582,9 @@ def checkout_form_validate(self, mode, all_form_values, data):
             if partner.exists() and partner.name and not partner.sudo().can_edit_vat() and 'name' in data and (data['name'] or False) != (partner.name or False):
                 error['name'] = 'error'
                 error_message.append(_('Changing your name is not allowed once invoices have been issued for your account. Please contact us directly for this operation.'))
+            if partner.exists() and partner.sudo().user_ids.employee_id.id:
```

Comment:
employee_id is not existing without hr.
Should check something related to field 'share' instead

Reply:
quid about commit msg... sale, account... while you touch website_sale ? 

Maybe the fix that block the checkout is bad idea, but just disallow to change name/email from the checkout for non shared user will be better. He will need to use /my/account if he want explicitly update his name.